tests/test_tensor/test_linear_tp.py

```diff
@@ -0,0 +1,110 @@
+import torch
+from colossalai.context.parallel_mode import ParallelMode
+from colossalai.tensor import ColoTensor
+from copy import deepcopy
+
+from functools import partial
+
+import colossalai
+import pytest
+import torch
+import torch.multiprocessing as mp
+from colossalai.logging import get_dist_logger
+from colossalai.testing import parameterize, rerun_if_address_is_in_use
+from colossalai.utils.cuda import get_current_device
+from colossalai.utils import free_port
+from colossalai.core import global_context as gpc
+import torch.distributed as dist
+
+def check_equal(A, B):
+    assert torch.allclose(A, B, rtol=1e-3, atol=1e-1) == True
+
+def run_linear_tp1d_row_test():
+    device = get_current_device()
+    dtype = torch.float32
+    DEPTH = 4
+    in_features = 4
+    out_features = 5
+
+    i = gpc.get_local_rank(ParallelMode.PARALLEL_1D)
+
+    layer = torch.nn.Linear(in_features, out_features)
+
+    A_shape = (2, in_features)
+    A_master = torch.randn(A_shape, dtype=dtype, device=device)
+    dist.broadcast(A_master, src=0)
+    A = A_master.clone()
+    A.requires_grad = True
+
+    W_shape = (out_features, in_features)
+    W_master = torch.randn(W_shape, dtype=dtype, device=device)
+    dist.broadcast(W_master, src=0)
+    W = torch.chunk(W_master, DEPTH, dim=-1)[i]
+    W = W.clone()
+    W.requires_grad = True
+
+    B_shape = (out_features)
+    B_master = torch.randn(B_shape, dtype=dtype, device=device)
+    dist.broadcast(B_master, src=0)
+    B = B_master.clone()
+    B.requires_grad = True
+
+    # replace the torch nn.Parameters with ShardedTensor
+    sharded_weight = ColoTensor.init_from_torch_tensor(W)
+    sharded_weight._shard_spec = "1Drow"
+    sharded_bias = ColoTensor.init_from_torch_tensor(B)
+    delattr(layer, 'weight')
+    setattr(layer, 'weight', sharded_weight)
+    delattr(layer, 'bias')
+    setattr(layer, 'bias', sharded_bias)
+    layer.weight.requires_grad = True
+    layer.bias.requires_grad = True
+    out = layer(A)
+
+    A_master = A_master.clone()
+    A_master.requires_grad = True
+    W_master = W_master.clone()
+    W_master.requires_grad = True
+    B_master = B_master.clone()
+    B_master.requires_grad = True
```

Comment:
why clone()?

Reply:
Just ensure the tesnors' clarity. I will remove them.